### PR TITLE
share graph object in TFNet

### PR DESCRIPTION
**Pull request by yangw1234**

### CODE REVIEW

**Comment by yangw1234** [comment]
manually tested on cluster

**Comment by yangw1234** [comment]
pr validation pass

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala [hunk 1]
```diff
@@ -232,6 +227,152 @@ class TFNet private(graphDef: Array[Byte],
...
+
+      val (graph, _) = getOrCreateGraph(id) {
+        val g = new Graph()
+        g.importGraphDef(graphDef)
+        g
+      }
+      tfGraph = graph
+    }
+
+    private var id = new Integer(tfGraph.hashCode())
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala]: how do we guarantee that id is always unique?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala [hunk 2]
```diff
@@ -232,6 +227,152 @@ class TFNet private(graphDef: Array[Byte],
...
+    }
+
+    private def writeInternal(out: CommonOutputStream): Unit = {
+      val (graphDef, _) = getOrCreateGraphDef(id) {
+        tfGraph.toGraphDef
+      }
+      val len = graphDef.length
+      out.writeInt(id)
+      out.writeInt(len)
+      out.write(graphDef)
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/net/TFNet.scala]: write the byte array out can still take a long time, yes?

**Pull request merged**